fhirstore/fhirstore.py

```diff
@@ -288,77 +286,124 @@ def search(
+                body=query, index=f"fhirstore.{search_args.resource_type.lower()}"
+            )
+
+            bundle.fill(hits, search_args.formatting_args)
 
         return bundle
 
-    def count(self, resource_type, params):
-        """Counts how many results match this query
+    def comprehensive_search(self, resource_type: str, args: ImmutableMultiDict):
+        """ To deal with keywords : _include, _revinclude, _has
         """
-        self.validate_resource_type(resource_type)
+        search_args = SearchArguments()
+        search_args.parse(args, resource_type)
+
+        # handle _has
+        rev_chain = search_args.reverse_chain
+        if rev_chain and rev_chain.is_queried == True:
+            inner_ids = []
+            # If there is a double _has chain
+            if len(rev_chain.has_args) == 2:
+                outer_ids = []
+                rev_args_outer = SearchArguments()
+
+                # parse the outer chain and search the store
+                rev_args_outer.parse(rev_chain.has_args[1], rev_chain.resources_type[1])
+                chained_bundle_outer = self.search(rev_args_outer)
+
+                for item in chained_bundle_outer.content["entry"]:
+                    outer_ids.append(
+                        item["resource"][rev_chain.references[1]]["reference"].split(
+                            sep="/", maxsplit=1
+                        )[1]
+                    )
 
-        core_query = build_core_query(params)
-        query = {"query": core_query}
-
-        # .lower() is used to fix the fact that monstache changes resourceTypes to
-        # all lower case
-        hits = self.es.count(body=query, index=f"fhirstore.{resource_type.lower()}")
-        return {
-            "resource_type": "Bundle",
-            "tag": {"code": "SUBSETTED"},
-            "total": hits["count"],
-        }
+                # fill the inner chain with the ids from the previous search
+                rev_chain.has_args[0][rev_chain.fields[0]] = outer_ids
+
+            # If there is a single _has chain, fill it with the value
+            elif len(rev_chain.has_args) == 1:
+                rev_chain.has_args[0][rev_chain.fields[0]] = rev_chain.value
+
+            rev_args_inner = SearchArguments()
+            rev_args_inner.parse(rev_chain.has_args[0], rev_chain.resources_type[0])
+            chained_bundle_inner = self.search(rev_args_inner)
+
+            for item in chained_bundle_inner.content["entry"]:
+                inner_ids.append(
+                    item["resource"][rev_chain.references[0]]["reference"].split(
+                        sep="/", maxsplit=1
+                    )[1]
+                )
+            search_args.core_args["id"] = inner_ids
+
+        bundle = self.search(search_args)
+
+        ## handle _include
+        if (
+            search_args.formatting_args["include"]
+            and search_args.formatting_args["is_summary_count"] == False
```

Comment:
`and not search_args.formatting_args["is_summary_count"]`

Reply:
done